### ISSUE TITLE
Bug with offsets/length for description in custom tags (and possibly its data)

### ISSUE THREAD

**Issue by silverspex** [Borewit/music-metadata#406]
This bug is probably related to what others have already reported, but I wanted to give an additional example. The offsets to where description resides seem to be calculated wrongly or using wrong encoding (?).

For example, a hex dump of a MP3 file containing custom Serato (a DJ software) ID3 frames - a small hex dump sample:

![tagsbug](https://user-images.githubusercontent.com/44753051/72149010-11830200-339a-11ea-9ed6-4624863453a0.jpg)

However, when parsed all of these ends up mangled for description etc. It's not consistent either so I guess the offset/length is read from the wrong position - but this is just an initial guess.

The output ends up something like this:

```javascript
[
  { id: 'TXXX:SERATO_PLAYCOUNT', value: '1' },
  {
    id: 'GEOB',
    value: {
      type: 'application/octet-stream',
      filename: '',
      description: 'erato Overview',    // Should be "Serato Overview", skipping 1 char
      data: <Buffer 65 72 61 74 6f ...  ---8X--- cut for brev.
    }
  },
  {
    id: 'GEOB',
    value: {
      type: 'application/octet-stream',
      filename: '',
      description: '',         // should have been "Serato Analysis", none...
      data: <Buffer 65 72 61 74 ...  ---8X--- cut for brev.
    }
  },
  {
    id: 'GEOB',
    value: {
      type: 'application/octet-stream',
      filename: '',
      description: 'erato Au',   //  "Serato Autotags", skipping one, length wrong
      data: <Buffer 65 72 61 74 ...   ---8X--- cut for brev.
    }
  },
  {
    id: 'GEOB',
    value: {
      type: 'application/octet-stream',
      filename: '',
      description: 'erato Markers_',   // "Serato Markers_", skipping one
      ...   ---8X--- etc.
```

**Comment by silverspex** [comment]
I have provided a test MP3 file which contains said tags, for testing:

https://raw.githubusercontent.com/silverspex/temp-test-serato-tag-bug/master/test.mp3

**Comment by Borewit** [comment]
Thank you for reporting the issue @silverspex .

> This bug is probably related to what others have already reported,

I does directly look to a known issue to me, have you seen the issue before?

> but I wanted to give an additional example

Samples demonstrating an issue are always welcome.

 Tomorrow, I will analyze the issue.

**Comment by Borewit** [comment]
It looks like the [ ID3v2.3 GEOB](http://id3.org/id3v2.3.0#General_encapsulated_object) is indeed decoded incorrect.

**Comment by Borewit** [comment]
Fixed in [v6.1.2](https://www.npmjs.com/package/music-metadata/v/6.1.2), thanks again for your detailed analysis.

**Comment by silverspex** [comment]
Thank you @Borewit for looking into this and fixing it so quickly!
